File: frontend/src/postbody.ts
import { Transaction, ItemTransaction, Item, StockIn, Project } from "./model";
import { ItemTransactionView } from "./view";

export interface TransactionPostBody {
  transaction: Transaction,
  itemTransactions: ItemTransactionView[],
  itemTransactionDeleteIds: number[]
}

export interface ItemPostBody {
  item: Item,
  withInitialStock: boolean,
  initialStockQty: number,
  project: Project | null
}

export interface StockInPostBody {
  item: Item, 
  stockIns: StockIn[]
}

File: postbody.go
package main

type LoginPostBody struct {
	Username string `json:"username"`
	Password string `json:"password"`
}

type PasswordPostBody struct {
	Password string `json:"password"`
}

type TransactionPostBody struct {
	Transaction              Transaction           `json:"transaction"`
	ItemTransactions         []ItemTransactionView `json:"itemTransactions"`
	ItemTransactionDeleteIds []uint                `json:"itemTransactionDeleteIds"`
}

type ItemPostBody struct {
	InitialStockQty  uint    `json:"initialStockQty"`
	Item             Item    `json:"item"`
	Project          Project `json:"project"`
	WithInitialStock bool    `json:"withInitialStock"`
}
